src/test/java/org/jabref/logic/citationkeypattern/CitationKeyGeneratorTest.java

```diff
@@ -1094,4 +1094,12 @@ void generateKeyCorrectKeyLengthWithAuthNofMthAndUnicode() {
 
         assertEquals(4, generateKey(bibEntry, "[auth4_1]").length());
     }
+
+    @Test
+    void generateKeyWithNonNormalizedUnicode() {
+        // https://github.com/JabRef/jabref/issues/6583
```

Comment:
I believe the issue does not need to be linked as the comment, but can be found via git log and blame... @koppor ?

Reply:
I think you are right on this, I've changed the test cases slightly in the latest commit.
I'd just like to leave a stronger "trail" as this is a patch on the unicode -> Western-* issue. I'll add the caveats to the top part of this PR instead.

Reply:
Issue linking should be avoided to have the code self-contained. Larger decisions should go into an ADR. Shorter decisions can go into the code. In case at hand, I just added the issue description as code comment: https://github.com/JabRef/jabref/pull/6938/files#diff-9d2fc955b825e69a852945899b4d62f3